controllers/backstage_deployment.go

```diff
@@ -312,3 +286,29 @@ func (r *BackstageReconciler) setDefaultDeploymentImage(deployment *appsv1.Deplo
 		}
 	})
 }
+
+func (r *BackstageReconciler) applyBackstageLabels(backstage bs.Backstage, deployment *appsv1.Deployment) {
+	setBackstageAppLabel(&deployment.Spec.Template.ObjectMeta.Labels, backstage)
+	setBackstageAppLabel(&deployment.Spec.Selector.MatchLabels, backstage)
+	r.labels(&deployment.ObjectMeta, backstage)
+}
+
+func (r *BackstageReconciler) applyApplicationParamsFromCR(backstage bs.Backstage, deployment *appsv1.Deployment) {
+	if backstage.Spec.Application != nil {
+		deployment.Spec.Replicas = backstage.Spec.Application.Replicas
+		if backstage.Spec.Application.Image != nil {
+			visitContainers(&deployment.Spec.Template, func(container *v1.Container) {
+				container.Image = *backstage.Spec.Application.Image
+			})
+		}
+
+		if len(backstage.Spec.Application.ImagePullSecrets) > 0 { // use image pull secrets from the CR spec
```

Comment:
What would happen if `backstage.Spec.Application.ImagePullSecrets` was non-empty, and then updated to an empty list? Shouldn't we update it in the Deployment as well?
I noticed that the Deployment still had the previous pull secret..

Reply:
Updated the PR so that in this case the deployment should be updated. Also as discussed, changed backstage.Spec.Application.ImagePullSecrets to a pointer so that we can distinguish between the default  (the pointer not set) and empty values for ImagePullSecrets.